silkworm/db/datastore/snapshots/snapshot_repository.cpp

```diff
@@ -32,13 +34,15 @@ namespace fs = std::filesystem;
 using namespace datastore;
 
 SnapshotRepository::SnapshotRepository(
+    RepositoryKind kind,
```

Comment:
let's pass an EntityName here
there are constants for each repo type: kBlocksRepositoryName, kStateRepositoryNameLatest, kStateRepositoryNameHistorical